stdlib/LinearAlgebra/src/cholesky.jl

```diff
@@ -174,6 +174,11 @@ function cholesky!(A::StridedMatrix, ::Val{false}=Val(false); check::Bool = true
     end
 end
 
+function cholesky(A::Diagonal, ::Val{false} = Val(false))
```

Comment:
You'd need to add the `check` keyword here and only throw when it is true and otherwise set the `info` value to the index of the first non-positive element.